lib/orderbook/OrderBook.ts

```diff
@@ -1,151 +1,212 @@
-        this.logger.debug(`order added: ${JSON.stringify(dbOrder)}`);
+        this.addOrder(this.ownOrders, remainingOrder);
+        this.logger.debug(`order added: ${JSON.stringify(remainingOrder)}`);
       }
     }
 
     return matchingResult;
   }
 
-  private addPeerOrder = async (order: orders.PeerOrder): Promise<void> => {
+  private addPeerOrder = (order: orders.PeerOrder) => {
     const matchingEngine = this.matchingEngines[order.pairId];
     if (!matchingEngine) {
       this.logger.debug(`incoming peer order invalid pairId: ${order.pairId}`);
       return;
     }
 
-    const stampedOrder: orders.StampedPeerOrder = { ...order, createdAt: new Date() };
+    const stampedOrder: orders.StampedPeerOrder = { ...order, createdAt: ms() };
     matchingEngine.addPeerOrder(stampedOrder);
-    const dbOrder = await this.repository.addOrder(stampedOrder);
-    this.logger.debug(`order added: ${JSON.stringify(dbOrder)}`);
+    this.addOrder(this.peerOrders, stampedOrder);
+    this.logger.debug(`order added: ${JSON.stringify(stampedOrder)}`);
+  }
+
+  private updateOrderQuantity = (type: { [pairId: string]: Orders }, order: orders.StampedOrder, decreasedQuantity: number) => {
+    const object = this.getOrderMap(type, order);
+    object[order.id].quantity = object[order.id].quantity - decreasedQuantity;
+    if (object[order.id].quantity === 0) {
+      delete object[order.id];
+    }
+  }
+
+  private addOrder = (type: { [pairId: string]: Orders }, order: orders.StampedOrder) => {
+    this.getOrderMap(type, order)[order.id] = order;
+  }
+
+  private getOrderMap = (type: { [pairId: string]: Orders }, order: orders.StampedOrder): { [ id: string ]: orders.StampedOrder } => {
+    const orders = type[order.pairId];
+    if (order.quantity > 0) {
+      return orders.buyOrders;
+    } else {
+      return orders.sellOrders;
+    }
+  }
+
+  private sendOrders = async (peer: Peer) => {
+    // TODO: just send supported pairs
+    const pairs = await this.getPairs();
+
+    const promises: Promise<orders.OutgoingOrder | void>[] = [];
+    for (const { id } of pairs) {
+      const orders = await this.getOwnOrders(id, 0);
+      Object.values(orders.buyOrders).forEach(order => promises.push(this.castToOutgoingOrder(order)));
+      Object.values(orders.sellOrders).forEach(order => promises.push(this.castToOutgoingOrder(order)));
+    }
+    await Promise.all(promises).then((outgoingOrders) => {
+      peer.sendOrders(outgoingOrders as orders.OutgoingOrder[]);
+    });
   }
 
   private broadcastOrder = async (order: orders.StampedOwnOrder): Promise<void> => {
-    if (!this.pool) {
-      return;
+    if (this.pool) {
+      const outgoingOrder = await this.castToOutgoingOrder(order);
+
+      if (outgoingOrder) {
+        this.pool.broadcastOrder(outgoingOrder);
+      }
     }
+  }
+
+  private castToOutgoingOrder = async (order: orders.StampedOwnOrder): Promise<orders.OutgoingOrder | void> => {
```

Comment:
I think that the name `castToOutgoingOrder` doesn't fit here, because you're not just casting, but also creating an invoice. So perhaps `createOutgoingOrder` is better (and also, maybe change `getInvoice` to `createInvoice`).